test/specs/product.page.test.js

```diff
@@ -0,0 +1,29 @@
+const { expect } = require("@wdio/globals");
+const ProductPage = require("../pageobjects/product.page");
+const HomePage = require("../pageobjects/home.page");
+const testData = require("../data/testData");
+
+describe("Product Page Tests", () => {
+  it("should select product size, color, set quantity, and add to cart", async () => {
+    const { size, color, quantity } = testData.product;
+
+    await HomePage.open();
+    await HomePage.searchForProduct("jacket");
+    await HomePage.openProductDetail(0);
+    
+    await ProductPage.productTitle.waitForDisplayed();
+    await ProductPage.selectProductSize(size);
+    await ProductPage.selectProductColor(color);
+    await ProductPage.setProductQuantity(quantity);
+    await ProductPage.addToCart();
```

Comment:
The name of the test can be 'Validate that we are able to choose a product.' All related methods can then be isolated in a ProductsUtility class, which will contain a selectAllProductInformation method with calls to each of these methods. After that, in this test, you’ll only need to call selectAllProductInformation(), and you can reuse this method in the smoke test as well without repeating the setup of this information.